### ISSUE TITLE
Support worktrees and submodules in hook auto-install

### ISSUE THREAD

**Issue by sbienkow-ninja** [ejoffe/spr#97]
Currently, running `git spr update` in either a worktree or a submodule, ends with:

```
2021/08/16 15:45:18 symlink /home/sbienkow/.local/bin/spr_commit_hook .git/hooks/commit-msg: not a directory
```

That is because in worktrees and submodules, `.git` is not a directory but a file which contains path to its ".git" folder.

It'd be to at least receive a more friendly message :)

**Comment by schreter** [comment]
+1 for worktree support.

FYI, in a worktree, `.git` is a file, which contains a single line `gitdir: <dir>`. In the referenced directory, there is contents similar to main `.git` directory, but not full, and additionally it contains a file `commondir` with the relative path to the primary `.git` directory (e.g., `../..` in my case). Thus, it's possible to find the actual `.git` directory relatively easily.

**Comment by ejoffe** [comment]
Fix released in v0.8.3 : thank you @mstory21
